mixer/adapter/solarwinds/config/config.proto

```diff
@@ -0,0 +1,131 @@
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     requestsize.metric.istio-system:
+//       name: request_size
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     responsesize.metric.istio-system:
+//       name: response_size
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     tcpbytesent.metric.istio-system:
+//       name: tcp_bytes_sent
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//     tcpbytereceived.metric.istio-system:
+//       name: tcp_bytes_received
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//   logs:
+//     solarwindslogentry.logentry.istio-system:
+//       payloadTemplate: '{{or (.originIp) "-"}} - {{or (.sourceUser) "-"}} [{{or (.timestamp.Format "2006-01-02T15:04:05Z07:00") "-"}}] "{{or (.method) "-"}} {{or (.url) "-"}} {{or (.protocol) "-"}}" {{or (.responseCode) "-"}} {{or (.responseSize) "-"}}'
+
+message Params {
+    // AppOptics Access Token needed to send metrics to AppOptics. If no access token is given then metrics
+    // will NOT be shipped to AppOptics
+    string appoptics_access_token = 1;
+
+    // Optional. Max batch size of metrics to be sent to AppOptics.
+    // AppOptics does not allow batch size greater than 1000.
+    // If this is not specified a default batch size of 1000 will be used.
+    int32 appoptics_batch_size = 2;
+
+    // Papertrail url to ship logs to. If no papertrail url is given then the logs will NOT be shipped but rather
+    // dropped.
+    string papertrail_url = 3;
+
+    // This is the duration for which logs will be persisted locally until it is shipped to papertrail in the event
+    // of a network failure
+    google.protobuf.Duration papertrail_local_retention_duration = 4 [(gogoproto.nullable)=true, (gogoproto.stdduration) = true];
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
+    message MetricInfo {
+        // The name is used to register the metric.
+        string name = 1;
+
+        // Optional. A human readable description of this metric.
+        string description = 2;
+
+        // The names of labels to use: these need to match the dimensions of the Istio metric.
+        repeated string label_names = 3;
+    }
+
+    // A map of Istio metric name to solarwinds metric info.
+    map<string, MetricInfo> metrics = 5;
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
+    message LogInfo {
+        // Optional. A golang text/template template that will be executed to construct the payload for this log entry.
```

Comment:
The typical operator is not go-literate, so "a golang text/template template" doesn't tell 'em much. Maybe supply a URL to a doc page that explains templates, and provide an example here?